tests/acceptance/bootstrap/SpacesContext.php

```diff
@@ -874,6 +918,32 @@ public function checkFileContent(
 		Assert::assertEquals($fileContent, $actualFileContent, "$file does not contain $fileContent");
 	}
 
+	/**
+	 * @Then /^for user "([^"]*)" the content of file "([^"]*)" of federated share "([^"]*)" should be "([^"]*)"$/
+	 *
+	 * @param string    $user
+	 * @param string    $file
+	 * @param string    $share
+	 * @param string    $fileContent
+	 *
+	 * @return void
+	 *
+	 * @throws Exception|GuzzleException
+	 */
+	public function checkFileContentOfRemoteShare(
+		string $user,
+		string $file,
+		string $share,
+		string $fileContent
+	): void {
+		// trimming the resource 'FOLDER/file.txt' to get the destination path 'file.txt'
+		// 'FOLDER' is represented by remoteItem id in the federation
+		$file = \ltrim($file, "/");
+
+		$actualFileContent = $this->getFileData($user, $share, $file, true)->getBody()->getContents();
+		Assert::assertEquals($fileContent, $actualFileContent, "$file does not contain $fileContent");
```

Comment:
I think this way, when a test fails, it is easy to get an idea about failure. In past, we faced similar problem  regarding insufficient detail on error message 
```suggestion
		Assert::assertEquals($fileContent, $actualFileContent, "Expected file to content to be $fileContent but found $actualFileContent");
```